File: Course Handicap Calculator/Utils/CourseHandicapCalculator.m
#import "CourseHandicapCalculator.h"


@implementation CourseHandicapCalculator

+ (NSInteger)courseHandicapWithHandicapIndex:(NSNumber *)handicapIndex courseSlope:(NSNumber *)courseSlope {
    double courseHandicapWithoutRound = handicapIndex.doubleValue * courseSlope.doubleValue / 113;
    
    NSInteger courseHandicap = lround(courseHandicapWithoutRound);
    
    return courseHandicap;
}

@end

File: Course Handicap Calculator/Utils/CourseHandicapCalculator.swift
import Foundation


class CourseHandicapCalculator {
    
    static func courseHandicapWithHandicapIndex(_ handicapIndex: Double?, courseSlope: Double?) -> Int {
        guard (handicapIndex != nil) && (courseSlope != nil) else {
            return 0
        }
    
        let courseHandicapWithoutRound = handicapIndex! * courseSlope! / 113
        
        let courseHandicap = lround(courseHandicapWithoutRound);
        return courseHandicap
    }
    
}
